Packs/GsuiteAuditor/Integrations/GsuiteAuditor/GsuiteAuditor.py

```diff
@@ -0,0 +1,344 @@
+        timedelta_param = timedelta(minutes=time_param)
+    elif time_list[1] == 'hours':
+        timedelta_param = timedelta(hours=time_param)
+    elif time_list[1] == 'days':
+        timedelta_param = timedelta(days=time_param)
+
+    end_time = dateparser.parse(start_time) + timedelta_param
+
+    return end_time.strftime(DATE_FORMAT)
+
+
+def fetch_request(client: GSuiteClient, app: str, start_time: str, end_time: str) -> List[Dict]:
+    """
+    for an app and start time, send an API request and returns the items
+    """
+    response = client.http_request(
+        url_suffix=URL_SUFFIX.format('all', app),
+        params={'startTime': start_time, 'endTime': end_time})
+
+    return response.get('items', [])
+
+
+@logger
+def test_module(client: GSuiteClient) -> str:
+    """
+    Performs test connectivity by valid http response
+
+    :param client: client object which is used to get response from api.
+
+    :return: raise ValueError if any error occurred during connection
+    :raises DemistoException: If there is any other issues while making the http call.
+    """
+    with GSuiteClient.http_exception_handler():
+        if ADMIN_EMAIL:
+            suffix = URL_SUFFIX.format('all', 'admin')
+            client.http_request(url_suffix=suffix, method='GET', params={'max_results': '1'})
+        else:
+            return_results("Please insert Admin Email parameter for the test to run")
+    return 'ok'
+
+
+''' COMMAND FUNCTIONS '''
+
+
+def activities_list_command(client: GSuiteClient, args: Dict[str, Any]) -> CommandResults:
+    """
+    Prints all activities in the G Suite instance.
+
+    :param client: client object which is used to get response from api
+    :param args: command arguments.
+
+    :return: CommandResults object with context and human-readable.
+    """
+    arguments = prepare_args_for_activities_list(args)
+    response = client.http_request(
+        url_suffix=URL_SUFFIX.format(args.get('user_key', 'all'), args.get('application_name')),
+        params=arguments)
+
+    # Readable Output
+    readable_items = prepare_readable_items(response.get('items', []))
+
+    readable_output = f'### Next Page Token: {response["nextPageToken"]}\n' if response.get("nextPageToken") else ''
+    readable_output += tableToMarkdown(
+        'Total Retrieved {}: {}'.format('Activities', len(response.get('items', []))),
+        readable_items,
+        ['time', 'applicationName', 'email', 'profileId', 'ipAddress', 'events']) if response.get(
+        'items') else 'No activities found for the given ' \
+                      'argument(s). '
+    # Context
+    context_outputs = prepare_output_for_activities_list(response)
+
+    return CommandResults(
+        outputs=context_outputs,
+        readable_output=readable_output,
+        raw_response=response
+    )
+
+
+@logger
+def fetch_incidents(client: GSuiteClient, first_fetch_time: str, fetch_limit: int, applications: Any) -> List[Dict]:
```

Comment:
this method is very long and complex, it might be easier to maintain if you do 2 things:
1) add dev comments about what you are doing in each section.
2) split some logic to sub-methods - like the setup or getting all the IDs that share the last time.